### ISSUE TITLE
Feature Request: Nested Route Guards

### ISSUE THREAD

**Issue by ajmeese7** [rkanik/sveltekit-route-guard#1]
Hey, super pumped about this library! I'm working on implementing it now as a low-overhead alternative to `svelte-routing`.

One thing I'd love to see for feature parity is nested routing, so multiple guards can easily be applied in the order of which they are encountered. For example, if my `/app/*` rule requires the user to be authenticated, but the `/app/dashboard` route also requires that the user have a specific role, I'd like to see those be applied in a nested manner. I think something like the following would be the optimal syntax to handle this:

```js
const routes = [
  {
    pathname: "/app/*",
    meta: {
      auth: true,
      types: [ UserType.Tenant ],
    },
    subroutes: [
      {
        pathname: "/app/dashboard",
        meta: {
          auth: true,
          roles: [ TenantUserRole.OWNER, TenantUserRole.ADMIN, TenantUserRole.MEMBER ],
        },
      },
    ],
  },
];

export const handle: Handle = createRouteGuard({
  redirect,
  routes,
  beforeEach(to, event, next) {
    ...
  }
});
```

I just don't want the `/app/*` route to capture all the matches and take priority over something more specific, if that makes sense. It wouldn't even have to have this fancy of a syntax, per-se, if it can jusat be made to evaluate rules with regex in them last. Not sure what the best way to implement this is, but I'd love to have the feature supported!

**Comment by rkanik** [comment]
Hello @ajmeese7, Thanks for submitting this issue, Actually I did it for my personal use, if you or anyone find it useful and want to participate to implement more features that you would be great. Do you want to implement that feature and create a PR.

**Comment by rkanik** [comment]
If we want to implement nested routes, then it should be like this,
pathname of parent should be concatenate with the children's. and also meta of parent will be extended to childs. See below in children we don't have to write `{ auth: true }`.

```ts
const routes = [
  {
    pathname: "/app",
    meta: {
      auth: true,
      types: [UserType.Tenant],
    },
    children: [
      {
        pathname: "/dashboard",
        meta: {
          roles: [TenantUserRole.ADMIN],
        },
      },
    ],
  },
];```

**Comment by ajmeese7** [comment]
@rkanik that example implementation looks perfect! Is that something you're willing to implement, or do I need to try to push up a PR?

**Comment by rkanik** [comment]
@ajmeese7 Yeah sure, please create a PR if you want to, otherwise I'll try to implement it in my free time(may be a weeks later).
